sympy/solvers/solveset.py

```diff
@@ -865,25 +866,20 @@ def solveset(f, symbol=None):
     elif not symbol.is_Symbol:
         raise ValueError('A Symbol must be given, not type %s: %s' % (type(symbol), symbol))
 
-    real = (symbol.is_real is True)
-
     f = sympify(f)
 
     if f is S.false:
         return EmptySet()
 
     if f is S.true:
-        if real:
-            return S.Reals
-        else:
-            return S.Complexes
+        return domain
 
     if isinstance(f, Eq):
         from sympy.core import Add
         f = Add(f.lhs, - f.rhs, evaluate=False)
 
     if f.is_Relational:
-        if real is False:
+        if not domain.is_subset(S.Reals):
             warnings.warn(filldedent('''
```

Comment:
Rather we should raise a `NotImplementedError` saying `Inequalities in the complex domain are not supported, try the real domain by setting domain=S.Reals`